### ISSUE TITLE
Globalnet pod in error state after joining clusters

### ISSUE THREAD

**Issue by manosnoam** [submariner-io/submariner#726]
After joining clusters, Globalnet pod status is _ERROR_ or _CrashLoopBackOff_.
This is a sporadic error, which happens once in a while on a new Submariner deployments:

https://qe-jenkins-csb-skynet.cloud.paas.psi.redhat.com/job/Submariner-OSP-AWS/755/Test-Report/
https://qe-jenkins-csb-skynet.cloud.paas.psi.redhat.com/job/debug_job/36/Test-Report/

**What happened**:

Deploying and joining clusters works as expected:

```
$ subctl deploy-broker  --service-discovery --kubecontext nmanos-cluster-a --globalnet

 • Deploying broker  ...
 ✓ Deploying broker
 • Creating broker-info.subm file  ...
 ✓ Creating broker-info.subm file
 ✓ Backed up previous broker-info.subm to broker-info.subm.2020-07-19T08_21_08+03_00
 ⚠ Reusing IPsec PSK from existing broker-info.subm

$ subctl join --clusterid nmanos-cluster-a   ./broker-info.subm    --ikeport 501 --nattport 4501 --version devel --subm-debug

* ./broker-info.subm says broker is at: https://api.nmanos-cluster-a.devcluster.openshift.com:6443
* There are 1 labeled nodes in the cluster:
  - ip-10-166-73-131.us-east-2.compute.internal
 • Discovering network details  ...
    Discovered network details:
        Network plugin:  OpenShift
        Service CIDRs:   [100.96.0.0/16]
        Cluster CIDRs:   [10.252.0.0/14]
 ✓ Discovering network details
 • Discovering multi cluster details  ...
 • Validating Globalnet configurations  ...
 ✓ Validating Globalnet configurations
 • Assigning Globalnet IPs  ...
 ✓ Assigning Globalnet IPs
 ✓ Allocated GlobalCIDR: 169.254.0.0/19
 ✓ Discovering multi cluster details
 • Deploying the Submariner operator  ...
 ✓ Deploying the Submariner operator
 ✓ Created operator CRDs
 ✓ Created operator namespace: submariner-operator
 ✓ Created operator service account and role
 ✓ Updated the privileged SCC
 ✓ Created lighthouse CRDs
 ✓ Created lighthouse service account and role
 ✓ Updated the privileged SCC
 ✓ Created Lighthouse service accounts and roles
 ✓ Deployed the operator successfully
 • Creating SA for cluster  ...
 ✓ Creating SA for cluster
 • Deploying Submariner  ...
 ✓ Deploying Submariner
 ✓ Submariner is up and running


$ subctl join --clusterid nmanos-cluster-b   ./broker-info.subm    --ikeport 501 --nattport 4501 --version devel --subm-debug

* ./broker-info.subm says broker is at: https://api.nmanos-cluster-a.devcluster.openshift.com:6443
* There are 1 labeled nodes in the cluster:
  - default-cl1-fqhg9-worker-jpp2k
 • Discovering network details  ...
    Discovered network details:
        Network plugin:  OpenShift
        Service CIDRs:   [100.96.0.0/16]
        Cluster CIDRs:   [10.252.0.0/14]
 ✓ Discovering network details
 • Discovering multi cluster details  ...
 • Validating Globalnet configurations  ...
 ✓ Validating Globalnet configurations
 • Assigning Globalnet IPs  ...
 ✓ Assigning Globalnet IPs
 ✓ Allocated GlobalCIDR: 169.254.32.0/19
 ✓ Discovering multi cluster details
 • Deploying the Submariner operator  ...
 ✓ Deploying the Submariner operator
 ✓ Created operator CRDs
 ✓ Created operator namespace: submariner-operator
 ✓ Created operator service account and role
 ✓ Created lighthouse CRDs
 ✓ Created lighthouse service account and role
 ✓ Created Lighthouse service accounts and roles
 ✓ Deployed the operator successfully
 • Creating SA for cluster  ...
 ✓ Creating SA for cluster
 • Deploying Submariner  ...
 ✓ Deploying Submariner
 ✓ Submariner is up and running
```

However, watching Submainer pods right after join completed - Globalnet pod is in "**Error**" state:

```
$ oc get all -n submariner-operator 

NAME                                                 READY   STATUS    RESTARTS   AGE
pod/submariner-gateway-s6ffm                         1/1     Running   0          21s
pod/submariner-globalnet-6k6lt                       0/1     Error     0          19s
pod/submariner-lighthouse-agent-6c787bddb5-rp52w     1/1     Running   0          18s
pod/submariner-lighthouse-coredns-7f4f69ddff-bq9jk   1/1     Running   0          18s
pod/submariner-lighthouse-coredns-7f4f69ddff-qrs6h   1/1     Running   0          18s
pod/submariner-operator-cfc8f4555-jv6cm              1/1     Running   0          43s
pod/submariner-routeagent-26f7l                      1/1     Running   0          20s
pod/submariner-routeagent-bf7qt                      1/1     Running   0          20s
pod/submariner-routeagent-lvrbj                      1/1     Running   0          20s
pod/submariner-routeagent-rgf94                      1/1     Running   0          20s

NAME                                    TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
service/submariner-lighthouse-coredns   ClusterIP   100.96.85.50    <none>        53/UDP              18s
service/submariner-operator-metrics     ClusterIP   100.96.147.31   <none>        8383/TCP,8686/TCP   24s

NAME                                   DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                AGE
daemonset.apps/submariner-gateway      1         1         1       1            1           submariner.io/gateway=true   21s
daemonset.apps/submariner-globalnet    1         1         0       1            0           submariner.io/gateway=true   19s
daemonset.apps/submariner-routeagent   4         4         4       4            4           <none>                       20s

NAME                                            READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/submariner-lighthouse-agent     1/1     1            1           18s
deployment.apps/submariner-lighthouse-coredns   2/2     2            2           18s
deployment.apps/submariner-operator             1/1     1            1           44s

NAME                                                       DESIRED   CURRENT   READY   AGE
replicaset.apps/submariner-lighthouse-agent-6c787bddb5     1         1         1       18s
replicaset.apps/submariner-lighthouse-coredns-7f4f69ddff   2         2         2       18s
replicaset.apps/submariner-operator-cfc8f4555              1         1         1       44s
```

**Here's Globalnet pod logs:**

```
oc -n submariner-operator describe pod submariner-globalnet-6k6lt
Name:               submariner-globalnet-6k6lt
Namespace:          submariner-operator
Priority:           0
PriorityClassName:  <none>
Node:               ip-10-166-73-131.us-east-2.compute.internal/10.166.73.131
Start Time:         Sun, 19 Jul 2020 08:21:41 +0300
Labels:             app=submariner-globalnet
                    component=globalnet
                    controller-revision-hash=679bc67699
                    pod-template-generation=1
Annotations:        openshift.io/scc: privileged
Status:             Running
IP:                 10.166.73.131
Controlled By:      DaemonSet/submariner-globalnet
Containers:
  submariner-globalnet:
    Container ID:   cri-o://39cf79c1bbffa5726d67f7ecf2e0856ecbe6d4c5f0fac1a770cb373d79cd3c2e
    Image:          quay.io/submariner/submariner-globalnet:devel
    Image ID:       quay.io/submariner/submariner-globalnet@sha256:1c17d9462e1527bdb7898b0e2874481e8fbe1dacb1b25724fd736ea2dd48a09f
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Sun, 19 Jul 2020 08:22:00 +0300
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Sun, 19 Jul 2020 08:21:59 +0300
      Finished:     Sun, 19 Jul 2020 08:21:59 +0300
    Ready:          True
    Restart Count:  1
    Environment:
      SUBMARINER_NAMESPACE:  submariner-operator
      SUBMARINER_CLUSTERID:  nmanos-cluster-a
      SUBMARINER_EXCLUDENS:  submariner-operator,kube-system,operators
    Mounts:
      /host from host-slash (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from submariner-operator-token-4h4r7 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  host-slash:
    Type:          HostPath (bare host directory volume)
    Path:          /
    HostPathType:  
  submariner-operator-token-4h4r7:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  submariner-operator-token-4h4r7
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  submariner.io/gateway=true
Tolerations:     node.kubernetes.io/disk-pressure:NoSchedule
                 node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/network-unavailable:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute
                 node.kubernetes.io/pid-pressure:NoSchedule
                 node.kubernetes.io/unreachable:NoExecute
                 node.kubernetes.io/unschedulable:NoSchedule
Events:
  Type    Reason     Age                From                                                  Message
  ----    ------     ----               ----                                                  -------
  Normal  Scheduled  <unknown>          default-scheduler                                     Successfully assigned submariner-operator/submariner-globalnet-6k6lt to ip-10-166-73-131.us-east-2.compute.internal
  Normal  Pulling    30s                kubelet, ip-10-166-73-131.us-east-2.compute.internal  Pulling image "quay.io/submariner/submariner-globalnet:devel"
  Normal  Pulled     15s                kubelet, ip-10-166-73-131.us-east-2.compute.internal  Successfully pulled image "quay.io/submariner/submariner-globalnet:devel"
  Normal  Pulled     14s                kubelet, ip-10-166-73-131.us-east-2.compute.internal  Container image "quay.io/submariner/submariner-globalnet:devel" already present on machine
  Normal  Created    13s (x2 over 14s)  kubelet, ip-10-166-73-131.us-east-2.compute.internal  Created container submariner-globalnet
  Normal  Started    13s (x2 over 14s)  kubelet, ip-10-166-73-131.us-east-2.compute.internal  Started container submariner-globalnet
(conda-test) /usr/bin/oc -n submariner-operator logs submariner-globalnet-6k6lt
+ trap 'exit 1' SIGTERM SIGINT
+ SUBMARINER_VERBOSITY=2
+ '[' '' == true ']'
+ DEBUG=-v=2
+ for f in iptables-save iptables
++ find_iptables_on_host iptables-save
++ chroot /host test -x /usr/sbin/iptables-save
++ echo /usr/sbin
++ return
iptables-save is present on the host at /usr/sbin/iptables-save
+ location=/usr/sbin
+ '[' /usr/sbin '!=' unknown ']'
+ echo 'iptables-save is present on the host at /usr/sbin/iptables-save'
+ sed 's!@@PATH@@!/usr/sbin!' /usr/sbin/iptables-wrapper.in
+ for f in iptables-save iptables
++ find_iptables_on_host iptables
++ chroot /host test -x /usr/sbin/iptables
iptables is present on the host at /usr/sbin/iptables
++ echo /usr/sbin
++ return
+ location=/usr/sbin
+ '[' /usr/sbin '!=' unknown ']'
+ echo 'iptables is present on the host at /usr/sbin/iptables'
+ sed 's!@@PATH@@!/usr/sbin!' /usr/sbin/iptables-wrapper.in
+ exec submariner-globalnet -v=2 -alsologtostderr
I0719 05:22:00.242197       1 main.go:33] Starting submariner-globalnet{nmanos-cluster-a [submariner-operator kube-system operators] submariner-operator []}
W0719 05:22:00.242431       1 client_config.go:549] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0719 05:22:00.263328       1 gatewaymonitor.go:81] Starting GatewayMonitor to monitor the active Gateway node in the cluster.
I0719 05:22:00.263346       1 gatewaymonitor.go:83] Waiting for informer caches to sync
I0719 05:22:00.363443       1 globalnet.go:140] Install/ensure SUBMARINER-GN-MARK chain exists
I0719 05:22:00.387946       1 gatewaymonitor.go:92] Starting endpoint worker.
I0719 05:22:00.394102       1 gatewaymonitor.go:126] In processNextEndpoint, endpoint info: {"metadata":{"name":"nmanos-cluster-b-submariner-cable-nmanos-cluster-b-10-166-0-229","namespace":"submariner-operator","selfLink":"/apis/submariner.io/v1/namespaces/submariner-operator/endpoints/nmanos-cluster-b-submariner-cable-nmanos-cluster-b-10-166-0-229","uid":"9de3c781-d708-4fb4-b81a-c4cab5aec36c","resourceVersion":"26846","generation":1,"creationTimestamp":"2020-07-19T05:22:00Z","managedFields":[{"manager":"submariner-engine","operation":"Update","apiVersion":"submariner.io/v1","time":"2020-07-19T05:22:00Z"}]},"spec":{"cluster_id":"nmanos-cluster-b","cable_name":"submariner-cable-nmanos-cluster-b-10-166-0-229","hostname":"default-cl1-fqhg9-worker-jpp2k","subnets":["169.254.32.0/19"],"private_ip":"10.166.0.229","public_ip":"66.187.233.202","nat_enabled":true,"backend":"strongswan"}}
I0719 05:22:00.394470       1 gatewaymonitor.go:128] Endpoint default-cl1-fqhg9-worker-jpp2k belongs to a remote cluster
I0719 05:22:00.394490       1 globalnet_egress.go:31] Marking traffic destined to remote cluster: -d 169.254.32.0/19 -j MARK --set-mark 0xC0000/0xC0000
I0719 05:22:00.407450       1 gatewaymonitor.go:126] In processNextEndpoint, endpoint info: {"metadata":{"name":"nmanos-cluster-a-submariner-cable-nmanos-cluster-a-10-166-73-131","namespace":"submariner-operator","selfLink":"/apis/submariner.io/v1/namespaces/submariner-operator/endpoints/nmanos-cluster-a-submariner-cable-nmanos-cluster-a-10-166-73-131","uid":"c026a9a4-128f-42ec-92bc-c3c093d88989","resourceVersion":"26841","generation":1,"creationTimestamp":"2020-07-19T05:21:59Z","managedFields":[{"manager":"submariner-engine","operation":"Update","apiVersion":"submariner.io/v1","time":"2020-07-19T05:21:59Z"}]},"spec":{"cluster_id":"nmanos-cluster-a","cable_name":"submariner-cable-nmanos-cluster-a-10-166-73-131","hostname":"ip-10-166-73-131","subnets":["169.254.0.0/19"],"private_ip":"10.166.73.131","public_ip":"18.220.202.197","nat_enabled":true,"backend":"strongswan"}}
I0719 05:22:00.407495       1 gatewaymonitor.go:158] We are now on GatewayNode 10.166.73.131
I0719 05:22:00.407512       1 gatewaymonitor.go:246] On Gateway Node, initializing ipamController.
I0719 05:22:00.413879       1 ipam.go:51] Setting up event handlers
I0719 05:22:00.413944       1 gatewaymonitor.go:261] Successfully initialized ipamController.
I0719 05:22:00.414371       1 ipam.go:87] Starting IPAM Controller
I0719 05:22:00.414388       1 globalnet.go:17] Install/ensure SUBMARINER-GN-INGRESS chain exists
I0719 05:22:00.458299       1 globalnet.go:27] Install/ensure SUBMARINER-GN-EGRESS chain exists
I0719 05:22:00.490985       1 globalnet.go:32] Install/ensure SUBMARINER-POSTROUTING chain exists
I0719 05:22:00.529144       1 globalnet.go:140] Install/ensure SUBMARINER-GN-MARK chain exists
I0719 05:22:00.600102       1 ipam.go:100] Waiting for informer caches to sync
I0719 05:22:00.700304       1 ipam.go:105] Starting workers
I0719 05:22:00.710494       1 ipam.go:557] Allocating globalIp 169.254.5.140 to Pod "openshift-kube-controller-manager/revision-pruner-8-ip-10-166-178-255.us-east-2.compute.internal" 
I0719 05:22:00.710524       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.252.0.58 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.5.140
E0719 05:22:00.712842       1 ipam.go:207] Node ip-10-166-73-131.us-east-2.compute.internal requeued 2 times
I0719 05:22:00.721394       1 globalnet.go:112] Skip processing "ip-10-166-154-129.us-east-2.compute.internal" node, as its a master node.
E0719 05:22:00.728839       1 ipam.go:207] Node ip-10-166-156-34.us-east-2.compute.internal requeued 2 times
E0719 05:22:00.733985       1 ipam.go:207] Node ip-10-166-170-225.us-east-2.compute.internal requeued 2 times
I0719 05:22:00.739427       1 globalnet.go:112] Skip processing "ip-10-166-178-255.us-east-2.compute.internal" node, as its a master node.
E0719 05:22:00.739742       1 ipam.go:207] service openshift-monitoring/cluster-monitoring-operator requeued 2 times
I0719 05:22:00.759770       1 ipam.go:332] In handleUpdateNode, node "ip-10-166-170-225.us-east-2.compute.internal" is not yet annotated with cniIfaceIP, enqueing
E0719 05:22:00.819992       1 ipam.go:207] Node ip-10-166-197-79.us-east-2.compute.internal requeued 2 times
I0719 05:22:00.862674       1 ipam.go:275] In handleUpdatePod, skipping pod "submariner-operator/submariner-globalnet-6k6lt" as it belongs to excluded namespace.
E0719 05:22:01.051496       1 ipam.go:207] service openshift-sdn/sdn requeued 2 times
I0719 05:22:01.219110       1 ipam.go:557] Allocating globalIp 169.254.5.176 to Pod "openshift-kube-scheduler/revision-pruner-8-ip-10-166-178-255.us-east-2.compute.internal" 
I0719 05:22:01.219142       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.252.0.57 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.5.176
I0719 05:22:01.419624       1 globalnet.go:112] Skip processing "ip-10-166-215-161.us-east-2.compute.internal" node, as its a master node.
I0719 05:22:01.640889       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-Z26MZGDJSJICLYJU" for service "openshift-service-ca-operator/metrics" now exists.
I0719 05:22:01.640922       1 ipam.go:510] Allocating globalIp 169.254.1.191 to Service "openshift-service-ca-operator/metrics" 
I0719 05:22:01.640941       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.1.191 -j KUBE-SVC-Z26MZGDJSJICLYJU
E0719 05:22:02.019491       1 ipam.go:207] Node ip-10-166-73-131.us-east-2.compute.internal requeued 3 times
I0719 05:22:02.419981       1 ipam.go:557] Allocating globalIp 169.254.6.53 to Pod "openshift-machine-config-operator/machine-config-operator-7974485c8b-9kd6w" 
I0719 05:22:02.420013       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.252.0.11 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.6.53
E0719 05:22:02.619027       1 ipam.go:207] Node ip-10-166-156-34.us-east-2.compute.internal requeued 3 times
I0719 05:22:02.842563       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-CIUYVLZDADCHPTYT" for service "openshift-apiserver-operator/metrics" now exists.
I0719 05:22:02.842597       1 ipam.go:510] Allocating globalIp 169.254.7.178 to Service "openshift-apiserver-operator/metrics" 
I0719 05:22:02.842616       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.7.178 -j KUBE-SVC-CIUYVLZDADCHPTYT
E0719 05:22:03.219901       1 ipam.go:207] Node ip-10-166-170-225.us-east-2.compute.internal requeued 4 times
I0719 05:22:03.619781       1 ipam.go:557] Allocating globalIp 169.254.28.252 to Pod "openshift-monitoring/alertmanager-main-2" 
I0719 05:22:03.619815       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.255.0.23 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.28.252
E0719 05:22:03.819341       1 ipam.go:207] Node ip-10-166-197-79.us-east-2.compute.internal requeued 3 times
I0719 05:22:04.040990       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-SSFS4UJOKJYBUN2S" for service "openshift-cloud-credential-operator/cco-metrics" now exists.
I0719 05:22:04.041025       1 ipam.go:510] Allocating globalIp 169.254.27.205 to Service "openshift-cloud-credential-operator/cco-metrics" 
I0719 05:22:04.041044       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.27.205 -j KUBE-SVC-SSFS4UJOKJYBUN2S
E0719 05:22:04.421026       1 ipam.go:207] Node ip-10-166-73-131.us-east-2.compute.internal requeued 4 times
I0719 05:22:04.820240       1 ipam.go:182] Ignoring pod "openshift-sdn/sdn-4wm7f" on host "10.166.178.255" as it uses hostNetworking
E0719 05:22:05.019343       1 ipam.go:207] Node ip-10-166-156-34.us-east-2.compute.internal requeued 4 times
I0719 05:22:05.246189       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-LR44LCGLBA5H46DK" for service "openshift-cluster-version/cluster-version-operator" now exists.
I0719 05:22:05.246221       1 ipam.go:510] Allocating globalIp 169.254.12.125 to Service "openshift-cluster-version/cluster-version-operator" 
I0719 05:22:05.246240       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.12.125 -j KUBE-SVC-LR44LCGLBA5H46DK
I0719 05:22:05.426697       1 ipam.go:182] Ignoring pod "openshift-cluster-machine-approver/machine-approver-6986df5bfb-vvqq7" on host "10.166.178.255" as it uses hostNetworking
E0719 05:22:05.619991       1 ipam.go:207] Node ip-10-166-170-225.us-east-2.compute.internal requeued 5 times
I0719 05:22:06.019197       1 ipam.go:557] Allocating globalIp 169.254.25.24 to Pod "openshift-etcd/installer-2-ip-10-166-178-255.us-east-2.compute.internal" 
I0719 05:22:06.019228       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.252.0.14 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.25.24
E0719 05:22:06.220031       1 ipam.go:207] Node ip-10-166-197-79.us-east-2.compute.internal requeued 4 times
I0719 05:22:06.442953       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-PCIHMO4L5A7KAMRN" for service "openshift-image-registry/image-registry" now exists.
I0719 05:22:06.442988       1 ipam.go:510] Allocating globalIp 169.254.15.221 to Service "openshift-image-registry/image-registry" 
I0719 05:22:06.443008       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.15.221 -j KUBE-SVC-PCIHMO4L5A7KAMRN
E0719 05:22:06.819690       1 ipam.go:207] Node ip-10-166-73-131.us-east-2.compute.internal requeued 5 times
I0719 05:22:07.223588       1 ipam.go:182] Ignoring pod "openshift-sdn/ovs-f8sk4" on host "10.166.170.225" as it uses hostNetworking
E0719 05:22:07.419092       1 ipam.go:207] Node ip-10-166-156-34.us-east-2.compute.internal requeued 5 times
E0719 05:22:07.641125       1 ipam.go:207] service openshift-image-registry/image-registry-operator requeued 2 times
I0719 05:22:07.819014       1 ipam.go:557] Allocating globalIp 169.254.3.145 to Pod "openshift-kube-controller-manager/installer-6-ip-10-166-154-129.us-east-2.compute.internal" 
I0719 05:22:07.819048       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.254.0.26 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.3.145
E0719 05:22:08.019730       1 ipam.go:207] Node ip-10-166-170-225.us-east-2.compute.internal requeued 6 times
E0719 05:22:08.240353       1 ipam.go:207] service openshift-monitoring/telemeter-client requeued 2 times
E0719 05:22:08.619359       1 ipam.go:207] Node ip-10-166-197-79.us-east-2.compute.internal requeued 5 times
I0719 05:22:08.841055       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-U3LVBEEPLKGG5GBK" for service "openshift-ingress/router-internal-default" now exists.
I0719 05:22:08.841086       1 ipam.go:510] Allocating globalIp 169.254.0.8 to Service "openshift-ingress/router-internal-default" 
I0719 05:22:08.841105       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.0.8 -j KUBE-SVC-U3LVBEEPLKGG5GBK
I0719 05:22:09.019809       1 ipam.go:557] Allocating globalIp 169.254.8.206 to Pod "openshift-kube-scheduler/installer-4-ip-10-166-154-129.us-east-2.compute.internal" 
I0719 05:22:09.019841       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.254.0.5 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.8.206
E0719 05:22:09.219976       1 ipam.go:207] Node ip-10-166-73-131.us-east-2.compute.internal requeued 6 times
E0719 05:22:09.819800       1 ipam.go:207] Node ip-10-166-156-34.us-east-2.compute.internal requeued 6 times
I0719 05:22:10.039868       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-OGQPOTBHHZMRDA43" for service "openshift-kube-scheduler/scheduler" now exists.
I0719 05:22:10.039903       1 ipam.go:510] Allocating globalIp 169.254.30.25 to Service "openshift-kube-scheduler/scheduler" 
I0719 05:22:10.039919       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.30.25 -j KUBE-SVC-OGQPOTBHHZMRDA43
I0719 05:22:10.220572       1 ipam.go:182] Ignoring pod "openshift-kube-apiserver/kube-apiserver-ip-10-166-154-129.us-east-2.compute.internal" on host "10.166.154.129" as it uses hostNetworking
E0719 05:22:10.420556       1 ipam.go:207] Node ip-10-166-170-225.us-east-2.compute.internal requeued 7 times
I0719 05:22:10.819860       1 ipam.go:557] Allocating globalIp 169.254.30.142 to Pod "openshift-dns/dns-default-bkvdk" 
I0719 05:22:10.819893       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.254.0.3 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.30.142
E0719 05:22:11.023431       1 ipam.go:207] Node ip-10-166-197-79.us-east-2.compute.internal requeued 6 times
I0719 05:22:11.241541       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-D5VYWAE3NWJS4H36" for service "openshift-kube-storage-version-migrator-operator/metrics" now exists.
I0719 05:22:11.241578       1 ipam.go:510] Allocating globalIp 169.254.15.60 to Service "openshift-kube-storage-version-migrator-operator/metrics" 
I0719 05:22:11.241601       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.15.60 -j KUBE-SVC-D5VYWAE3NWJS4H36
E0719 05:22:11.619465       1 ipam.go:207] Node ip-10-166-73-131.us-east-2.compute.internal requeued 7 times
I0719 05:22:12.019949       1 ipam.go:557] Allocating globalIp 169.254.0.171 to Pod "openshift-image-registry/image-registry-7f754b7dc-qk6pp" 
I0719 05:22:12.019983       1 globalnet_egress.go:15] Installing iptable egress rules for Pod: -p all -s 10.255.0.15 -m mark --mark 0xC0000/0xC0000 -j SNAT --to 169.254.0.171
E0719 05:22:12.219470       1 ipam.go:207] Node ip-10-166-156-34.us-east-2.compute.internal requeued 7 times
I0719 05:22:12.442203       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-G5A7ID5ATXHWKRS5" for service "openshift-monitoring/thanos-querier" now exists.
I0719 05:22:12.442238       1 ipam.go:510] Allocating globalIp 169.254.1.31 to Service "openshift-monitoring/thanos-querier" 
I0719 05:22:12.442264       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.1.31 -j KUBE-SVC-G5A7ID5ATXHWKRS5
E0719 05:22:12.821027       1 ipam.go:207] Node ip-10-166-170-225.us-east-2.compute.internal requeued 8 times
I0719 05:22:13.219480       1 ipam.go:182] Ignoring pod "openshift-monitoring/node-exporter-c247m" on host "10.166.215.161" as it uses hostNetworking
E0719 05:22:13.420531       1 ipam.go:207] Node ip-10-166-197-79.us-east-2.compute.internal requeued 7 times
I0719 05:22:13.639854       1 globalnet.go:95] kube-proxy chain "KUBE-SVC-LS7JF6SL4ODP2YA4" for service "openshift-cloud-credential-operator/controller-manager-service" now exists.
I0719 05:22:13.639883       1 ipam.go:510] Allocating globalIp 169.254.5.91 to Service "openshift-cloud-credential-operator/controller-manager-service" 
I0719 05:22:13.639900       1 globalnet_ingress.go:17] Installing iptables rule for Service -d 169.254.5.91 -j KUBE-SVC-LS7JF6SL4ODP2YA4
I0719 05:22:13.819693       1 ipam.go:182] Ignoring pod "openshift-monitoring/node-exporter-nm55r" on host "10.166.178.255" as it uses hostNetworking
E0719 05:22:14.019262       1 ipam.go:207] Node ip-10-166-73-131.us-east-2.compute.internal requeued 8 times
```

**Environment**:
- subctl version: v0.5.0-rc0-1-g67885f5

- Cluster A (AWS public)
Client Version: 4.5.2
Server Version: 4.5.2
Kubernetes Version: v1.18.3+b74c5ed

- Cluster B (OSP on-premise)
Client Version: 4.5.2
Server Version: 4.4.3
Kubernetes Version: v1.17.1

**Comment by sridhargaddam** [comment]
Thanks for reporting the issue @manosnoam 

The logs seem to correspond to a successful run. Also, it is not clear from the above logs if the Pod went into "Error" state due to Globalnet-issue or non-globalnet-issue (i.e., error while downloading image, or transient errors while bringing up the pod etc).

  submariner-globalnet:
    Container ID:   cri-o://39cf79c1bbffa5726d67f7ecf2e0856ecbe6d4c5f0fac1a770cb373d79cd3c2e
    Image:          quay.io/submariner/submariner-globalnet:devel
    Image ID:       quay.io/submariner/submariner-globalnet@sha256:1c17d9462e1527bdb7898b0e2874481e8fbe1dacb1b25724fd736ea2dd48a09f
    Port:           <none>
    Host Port:      <none>
    **State:          Running**
      Started:      Sun, 19 Jul 2020 08:22:00 +0300
    **Last State:     Terminated
      Reason:       Error**
      Exit Code:    255
      Started:      Sun, 19 Jul 2020 08:21:59 +0300
      Finished:     Sun, 19 Jul 2020 08:21:59 +0300
    Ready:          True
    Restart Count:  1

Next time when you observe this issue, can you please capture the output of the following command which will include some additional information on the state transitions.
`kubectl get pod submariner-globalnet-6k6lt -n submariner-operator -o yaml`

Note: If this issue is seen only during the initial deployment, we can consider this as a lower priority bug.

**Comment by sridhargaddam** [comment]
Okay, I was able to reproduce this issue locally by installing ONLY submariner-globalnet pod and disabling other Pods (particularly submariner-gateway pod).

**Basically the following is happening:** 
During the installation phase (aka when join command is issued), typically all the Submariner pods would be initialized in parallel. Submariner-gateway Pod, being one of them, is responsible for creating the clusterCRD and the creation of this CRD would normally take a few seconds. In the meantime, if GlobalnetPod tries to read the clusterCRD[1] and finds it missing, it simply exits with an appropriate message (as shown in the logs below).
Once the GlobalnetPod is restarted, it usually finds the ClusterCRD (as submariner-gateway pod would have been initialized by then and created the clusterCRD) and continues to Run normally. IMHO this is in-line with the expected behavior and not an issue as this could happen ONLY during installation and not at run-time. 

CC: @nyechiel  @mangelajo @tpantelis @vthapar @skitt 

[1] https://github.com/submariner-io/submariner/blob/d8ec1970b1dca95c0d0049d59a1357871facbf62/pkg/globalnet/main.go#L50

```
[sgaddam@localhost submariner]$ kubectl -n submariner-operator logs submariner-globalnet-htv7k
+ trap 'exit 1' SIGTERM SIGINT
+ SUBMARINER_VERBOSITY=2
+ '[' '' == true ']'
+ DEBUG=-v=2
+ for f in iptables-save iptables
++ find_iptables_on_host iptables-save
++ chroot /host test -x /usr/sbin/iptables-save
++ echo /usr/sbin
++ return
+ location=/usr/sbin
+ '[' /usr/sbin '!=' unknown ']'
iptables-save is present on the host at /usr/sbin/iptables-save
+ echo 'iptables-save is present on the host at /usr/sbin/iptables-save'
+ sed 's!@@PATH@@!/usr/sbin!' /usr/sbin/iptables-wrapper.in
+ for f in iptables-save iptables
++ find_iptables_on_host iptables
++ chroot /host test -x /usr/sbin/iptables
++ echo /usr/sbin
++ return
+ location=/usr/sbin
+ '[' /usr/sbin '!=' unknown ']'
+ echo 'iptables is present on the host at /usr/sbin/iptables'
iptables is present on the host at /usr/sbin/iptables
+ sed 's!@@PATH@@!/usr/sbin!' /usr/sbin/iptables-wrapper.in
+ exec submariner-globalnet -v=2 -alsologtostderr
I0720 15:43:48.109169       1 main.go:33] Starting submariner-globalnet{cluster2 [submariner-operator kube-system operators] submariner-operator []}
W0720 15:43:48.109312       1 client_config.go:549] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
F0720 15:43:48.115187       1 main.go:50] error while retrieving the local cluster "cluster2" info: clusters.submariner.io "cluster2" not found
```

**Comment by manosnoam** [comment]
Thanks @sridhargaddam 

If there's an option to change the logic, so it would not show Error on Globalnet pod during installation, it will be great.

In my experience, when later, users look for error causes on any issue, they may find this false Error alert, if digging in pods logs.

If this is too complected/overhead to fix, we can ignore it.

**Comment by sridhargaddam** [comment]
There are a few things we can do

1. Modify subctl/operator to install Globalnet only after Submariner Gateway initializes.
2. Modify Globalnet Controller to listen for ClusterCRD events and initialize itself only after one is present. Unfortunately, this is not a trivial change and has to be done carefully. If we want to pursue this approach IMHO we can keep this issue open and treat this issue as a low-priority issue.
3. Modify Globalnet Controller to retry ClusterCRD info a couple of times (for few seconds) and if it's still missing, exit and log an error message (similar to what we are doing today). 
4. As this is not seen frequently (TBH, I've never seen this in KIND setup) and when this happens, the "Age" difference between GlobalnetPod and GatewayPod would be having a difference of 2 or 3 secs as can be seen in the logs shared earlier. Document this behavior as part of the troubleshooting guide. 

```
$ oc get all -n submariner-operator 
NAME                                                 READY   STATUS    RESTARTS   AGE
pod/submariner-gateway-s6ffm                         1/1     Running   0          21s
pod/submariner-globalnet-6k6lt                       0/1     Error     0          19s
```

If others have any suggestions, please comment.

**Comment by sridhargaddam** [comment]
I'm in **favor of Option3** above, as its easy to implement, will not increase startup time during happy-path and less chances for any regressions.

**Comment by manosnoam** [comment]
Thanks @sridhargaddam 
The Globalnet pod error state has not appeared any more, since your fixed was merged. 
Closing this bug.
